Extension/src/LanguageServer/configurations.ts

```diff
@@ -737,33 +738,57 @@ export class CppProperties {
 
     private resolveAndSplit(paths: string[] | undefined, defaultValue: string[] | undefined, env: Environment, glob = false): string[] {
         const result: string[] = [];
-        if (paths) {
-            paths = this.resolveDefaults(paths, defaultValue);
-            paths.forEach(entry => {
-                const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
-                result.push(...entries);
-            });
-        }
-        if (!glob) {
+        if (paths === undefined) {
             return result;
         }
-
-        const globResult: string[] = [];
-        for (let res of result) {
-            // fastGlob will expand the ending double wildcard. temporary strip them before expanding
-            const recursive: boolean = res.endsWith('**');
-            if (recursive) {
-                res = res.slice(0, res.length - 2);
+        paths = this.resolveDefaults(paths, defaultValue);
+        paths.forEach(entry => {
+            const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
+            if (!entry.includes('*')){
```

Comment:
This `if (!entry.includes('*')){` seems incorrect (e.g. if glob is false, then the entries with ** are removed). I think that check should be removed and it should iterate through `result` instead of `paths` below.

Reply:
This is not the case if you are referring to examples like "sources/**".  I would suggest testing it. Please let me know if you find a bug repro and I will fix it.